### PR TITLE
Support incremental compilation

### PR DESCRIPTION
**Pull request by mdesmet**
### Problem

Support compilation preview and query preview of incremental models

### Solution

Describe the implemented solution. Add external references if needed.

### Screenshot/Demo

A picture is worth a thousand words. Please highlight the changes if applicable.

### How to test

- Steps to be followed to verify the solution or code changes
- Mention if there is any settings configuration added/changed/deleted

## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

### CODE REVIEW

File: src/content_provider/sqlPreviewContentProvider.ts [hunk 1]
```diff
@@ -78,15 +79,17 @@ export class SqlPreviewContentProvider
...
       const project = this.dbtProjectContainer.findDBTProject(Uri.file(fsPath));
       if (project === undefined) {
         this.telemetry.sendTelemetryError("sqlPreviewNotLoadingError");
         return "Still loading dbt project, please try again later...";
       }
       this.telemetry.sendTelemetryEvent("requestCompilation");
-      await project.refreshProjectConfig();
-      const result = await project.unsafeCompileQuery(query);
+      // TODO: figure out why this one needs to be commented
+      // await project.refreshProjectConfig();
```
Comment (mdesmet) [inline src/content_provider/sqlPreviewContentProvider.ts:91]: This line seems to cause issues even in master for models with macros.

File: dbt_core_integration.py [hunk 2]
```diff
@@ -579,10 +583,20 @@ def get_source_node(
...
-    def get_server_node(self, sql: str, node_name="name"):
+    def get_server_node(self, sql: str, node_name="name", original_node: Optional[Union["ManifestNode", str]] = None):
         """Get a node for SQL execution against adapter"""
         self._clear_node(node_name)
         sql_node = self.sql_parser.parse_remote(sql, node_name)
+        # Enable copying original node properties
+        if original_node is not None:
+            if isinstance(original_node, str):
+                original_node = self.get_ref_node(original_node)
+            if original_node.node_info["materialized"] == "incremental":
```
Comment (suryaiyer95) [inline dbt_core_integration.py:594]: For a node type, can it give key error?

File: dbt_core_integration.py [hunk 3]
```diff
@@ -371,6 +371,15 @@ def init_config(self):
         self.config = RuntimeConfig.from_args(self.args)
         if hasattr(self.config, "source_paths"):
             self.config.model_paths = self.config.source_paths
+        
+    def create_parser(self) -> None:
```
Comment (suryaiyer95) [inline dbt_core_integration.py:375]: WHy moved up?

**Review by saravmajestic — APPROVED** [review]

**Comment by anandgupta42** [comment]
Released in version 0.40.0

**Pull request merged**